Libraries/MiscDrivers/Camera/gc0308.c

```diff
@@ -0,0 +1,758 @@
+/******************************************************************************
+ *
+ * Copyright 2023 Analog Devices, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include <math.h>
+#include "camera.h"
+#include "sccb.h"
+#include "mxc_delay.h"
+#include "mxc_device.h"
+#include "gc0308_regs.h"
+
+#include "board.h"
+/* Note: board.h is included here because the GC0308 uses a hardware pin
+to toggle sleep mode.  "Camera_Sleep" should be implemented in the BSP, since
+the pin routing is board-specific.
+*/
+
+#define cambus_write(addr, x) sccb_write_byt(g_slv_addr, addr, x)
+#define cambus_read(addr, x) sccb_read_byt(g_slv_addr, addr, x)
+
+#define ORGN
+
+#define MSB(val) (((val) >> 8) & 0xFF)
+#define LSB(val) ((val)&0xFF)
+
+#ifndef PRINTF_DEBUG
+#ifdef DEBUG
+#define PRINTF_DEBUG(...) printf(__VA_ARGS__);
+#else
+#define PRINTF_DEBUG(...)
+#endif
+#endif
+
+static const uint8_t default_regs[][2] = {
+    { BANKSEL_RESET, 0x80 }, // Reset Chip
+    { BANKSEL_RESET, 0x00 }, // Release from reset and select Page-0
+    { HORIZONTAL_BLANKING, 0x6a },
+    /* If you see a bar on the left side of
+        the image you need to tweak (probably
+        increase) the horizontal blanking.
+    */
+
+    { EXPOSURE_MSB, 0x01 }, // Set exposure time
+    { EXPOSURE_LSB, 0x2c },
+    { AAAA_ENABLE, 0x06 }, // Enable AWB and ABS
+
+    { 0x0f, 0x00 },
+    { 0xe2, 0x00 }, //anti-flicker step [11:8]
+    { 0xe3, 0x96 }, //anti-flicker step [7:0]
+    { 0xe4, 0x01 }, //exp level 1  16.67fps
+    { 0xe5, 0xfe },
+    { 0xe6, 0x02 }, //exp level 2  12.5fps
+    { 0xe7, 0x58 },
+    { 0xe8, 0x02 }, //exp level 3  8.33fps
+    { 0xe9, 0x58 },
+    { 0xea, 0x02 }, //exp level 4  4.00fps
+    { 0xeb, 0x58 },
+    { 0xec, 0x00 },
```

Comment:
Did we take these camera drivers from somewhere too?

Reply:
Not as far as I'm aware.  We were provided some app notes and defaults by GalaxyCore that most of these settings are derived from, and then I did an optimization pass and modified them further